src/sinks/datadog/events/service.rs

```diff
@@ -37,16 +40,27 @@ impl DriverResponse for DatadogEventsResponse {
             output: None,
         }
     }
+
+    fn bytes_sent(&self) -> Option<BytesSent> {
+        Some(BytesSent {
+            byte_size: self.raw_byte_size,
+            protocol: &self.protocol,
+        })
+    }
 }
 
 #[derive(Clone)]
 pub struct DatadogEventsService {
+    protocol: String,
     batch_http_service:
         HttpBatchService<Ready<Result<http::Request<Bytes>, crate::Error>>, DatadogEventsRequest>,
 }
 
 impl DatadogEventsService {
     pub fn new(endpoint: String, default_api_key: String, http_client: HttpClient<Body>) -> Self {
+        let uri = http::Uri::try_from(endpoint.clone()).expect("URI not valid");
```

Comment:
Is the `endpoint` already validated by this point? If so, could we change the signature for this to take an actual `Uri`? Just want to make sure Vector is returning an error rather than panicking if users provide an invalid endpoint.